### PR TITLE
Update installation options for release-1.2

### PR DESCRIPTION
**Pull request by johnma14**
Update the installation options for release-1.2 and the installation
option changes between release 1.1 and release 1.2. These changes have
been generated by the updated changes to tablegen.py in the
PR: https://github.com/istio/istio.io/pull/4402

Fixes: #4374

### CODE REVIEW

File: content/docs/reference/config/installation-options-changes/index.md [hunk 1]
```diff
@@ -17,443 +17,237 @@ The tables below show changes made to the installation options used to customize
...
-| Key | Old Default Value | New Default Value | Old Description | New Description |
-| --- | --- | --- | --- | --- |
-| `certmanager.tag` | `v0.3.1` | `v0.6.2` |  |  |
-
-### Modified `kiali` key/value pairs
-
-| Key | Old Default Value | New Default Value | Old Description | New Description |
-| --- | --- | --- | --- | --- |
-| `kiali.tag` | `istio-release-1.0` | `v0.14` |  |  |
+| `global.tag` | `release-1.1-latest-daily` | `master-latest-daily` | `Default tag for Istio images.`  | `Default tag for Istio images.` |
```
Comment (linsun) [inline content/docs/reference/config/installation-options-changes/index.md]: Is it possible this is run against master?  Ideally, it would be run against 1.1 to 1.2

Also, i wonder if we should even show tag changes for istio core components, especially the components who are versioned the same way as istio.

File: content/docs/reference/config/installation-options-changes/index.md [hunk 2]
```diff
@@ -17,443 +17,237 @@ The tables below show changes made to the installation options used to customize
...
+| `global.proxy.resources.limits.memory` | `128Mi` | `1024Mi` |  |  |
```
Comment (linsun) [inline content/docs/reference/config/installation-options-changes/index.md]: @jnativio @mandarjog can we call this out in release note with an explanation pls?  

cc @duderino @francoispqt

File: content/docs/reference/config/installation-options-changes/index.md [hunk 3]
```diff
@@ -17,443 +17,237 @@ The tables below show changes made to the installation options used to customize
...
+| `global.proxy.dnsRefreshRate` | `5s` | `300s` | `Configure the DNS refresh rate for Envoy cluster of type STRICT_DNS 5 seconds is the default refresh rate used by Envoy`  | `Configure the DNS refresh rate for Envoy cluster of type STRICT_DNS This must be given it terms of seconds. For example, 300s is valid but 5m is invalid.` |
```
Comment (linsun) [inline content/docs/reference/config/installation-options-changes/index.md]: @howardjohn do you recall why this change is made?

File: content/docs/reference/config/installation-options-changes/index.md [hunk 4]
```diff
@@ -17,443 +17,237 @@ The tables below show changes made to the installation options used to customize
...
-| `mixer.telemetry.resources.limits.memory` | `4G` |  |
-| `mixer.podAnnotations` | `{}` |  |
-| `mixer.nodeSelector` | `{}` |  |
-| `mixer.adapters.kubernetesenv.enabled` | `true` |  |
-| `mixer.adapters.stdio.enabled` | `false` |  |
-| `mixer.adapters.stdio.outputAsJson` | `true` |  |
-| `mixer.adapters.prometheus.enabled` | `true` |  |
-| `mixer.adapters.prometheus.metricsExpiryDuration` | `10m` |  |
-| `mixer.adapters.useAdapterCRDs` | `true` | `Setting this to false sets the useAdapterCRDs mixer startup argument to false` |
+| `mixer.telemetry.reportBatchMaxEntries` | `100` | `Set reportBatchMaxEntries to 0 to use the default batching behavior (i.e., every 100 requests). A positive value indicates the number of requests that are batched before telemetry data is sent to the mixer server` |
```
Comment (linsun) [inline content/docs/reference/config/installation-options-changes/index.md]: I thought this change is only for master, not in 1.2...cc @huang195

File: content/docs/reference/config/installation-options-changes/index.md (hunk 3 above)
Comment (howardjohn) [reply content/docs/reference/config/installation-options-changes/index.md]: Yes. 5s means that for every DNS ServiceEntry you make (+2 for pilot and tracing), a DNS request will be made, for each proxy. And due to how kube-dns works generally a dns request actually becomes 6 requests for external urls. So people were having their DNS overloaded.

https://github.com/istio/istio/issues/13710 for details

No one should need to configure it now that 5m is the default, and we will replace it with proper TTL support soon

File: content/docs/reference/config/installation-options-changes/index.md (hunk 1 above)
Comment (johnma14) [reply content/docs/reference/config/installation-options-changes/index.md]: I will go ahead and push a change with the comparison against the 1.2 branch. I was running this against master. Thanks Lin.

File: content/docs/reference/config/installation-options-changes/index.md (hunk 4 above)
Comment (johnma14) [reply content/docs/reference/config/installation-options-changes/index.md]: @linsun @huang195 I will run this against the 1.2 branch and can verify this again.

**Comment by johnma14** [comment]
@sdake @linsun @howardjohn I just pushed a new change for the installation-options and installation-option-changes to reflect installation options in the release-1.2.0-rc-2.

File: content/docs/reference/config/installation-options-changes/index.md [hunk 5]
```diff
@@ -17,443 +17,225 @@ The tables below show changes made to the installation options used to customize
...
-| Key | Old Default Value | New Default Value | Old Description | New Description |
-| --- | --- | --- | --- | --- |
-| `certmanager.tag` | `v0.3.1` | `v0.6.2` |  |  |
-
-### Modified `kiali` key/value pairs
-
-| Key | Old Default Value | New Default Value | Old Description | New Description |
-| --- | --- | --- | --- | --- |
-| `kiali.tag` | `istio-release-1.0` | `v0.14` |  |  |
+| `global.hub` | `gcr.io/istio-release` | `docker.io/istio` | `Default hub for Istio images. Releases are published to docker hub under 'istio' project. Daily builds from prow are on gcr.io, and nightly builds from circle on docker.io/istionightly`  | `Default hub for Istio images. Releases are published to docker hub under 'istio' project. Daily builds from prow are on gcr.io, and nightly builds from circle on docker.io/istionightly` |
```
[the remark this answers is not in the document]
Comment (johnma14) [reply content/docs/reference/config/installation-options-changes/index.md]: @howardjohn I can filter this one out.

File: content/docs/reference/config/installation-options-changes/index.md [hunk 6]
```diff
@@ -17,443 +17,225 @@ The tables below show changes made to the installation options used to customize
...
-| `gateways.istio-ilbgateway.nodeSelector` | `{}` |  |
+| `gateways.istio-ingressgateway.sds.resources.requests.cpu` | `100m` |  |
+| `gateways.istio-ingressgateway.sds.resources.requests.memory` | `128Mi` |  |
+| `gateways.istio-ingressgateway.sds.resources.limits.cpu` | `2000m` |  |
+| `gateways.istio-ingressgateway.sds.resources.limits.memory` | `1024Mi` |  |
+| `gateways.istio-ingressgateway.resources.requests.cpu` | `100m` |  |
+| `gateways.istio-ingressgateway.resources.requests.memory` | `128Mi` |  |
+| `gateways.istio-ingressgateway.resources.limits.cpu` | `2000m` |  |
+| `gateways.istio-ingressgateway.resources.limits.memory` | `1024Mi` |  |
+| `gateways.istio-ingressgateway.applicationPorts` | `""` |  |
```
[the remark this answers is not in the document]
Comment (johnma14) [reply content/docs/reference/config/installation-options-changes/index.md]: @howardjohn Extracting the descriptions correctly has been a tricky thing to do. In the code, I do a lot of checks to filter out descriptions to make sure we get the right descriptions for the right configuration option. For example, in some of the values.yaml, old configuration options have been commented out and if they appear right before the next configuration option, without those filters in place, it will consider those commented out options as comments for it. So I think if the comment line contains the format: <something>: <something>, I just ignore it. The thinking was it was better to not have a description vs putting the wrong description for a configuration option. I will go ahead and verify this one more time and see if there is anything I can do to get this description. @ostromart was working on creating a more robust system to work as a schema for these configuration options which is why we did not go down the route of cleaning up all these values.yaml to remove the commented out configuration options.

File: content/docs/reference/config/installation-options-changes/index.md [hunk 7]
```diff
@@ -17,443 +17,225 @@ The tables below show changes made to the installation options used to customize
...
 | Key | Default Value | Description |
 | --- | --- | --- |
-| `global.hyperkube.tag` | `v1.7.6_coreos.0` |  |
-| `global.k8sIngressHttps` | `false` |  |
-| `global.crds` | `true` |  |
-| `global.hyperkube.hub` | `quay.io/coreos` |  |
-| `global.meshExpansion` | `false` |  |
-| `global.k8sIngressSelector` | `ingress` |  |
-| `global.meshExpansionILB` | `false` |  |
+| `global.proxy.tracer` | `"zipkin"` | `Specify which tracer to use. One of: lightstep, zipkin`  |
```
[the remark this answers is not in the document]
Comment (johnma14) [reply content/docs/reference/config/installation-options-changes/index.md]: @howardjohn I know why this is and will push a fix to resolve this. Thanks.

File: content/docs/reference/config/installation-options-changes/index.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (howardjohn) [reply content/docs/reference/config/installation-options-changes/index.md]: Ok sounds good! just wasnted sure why it was happening, the `:` makes sense. Thanks

File: content/docs/reference/config/installation-options-changes/index.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (linsun) [reply content/docs/reference/config/installation-options-changes/index.md]: +1 thanks Mariam

File: content/docs/reference/config/installation-options-changes/index.md (hunk 3 above)
Comment (linsun) [reply content/docs/reference/config/installation-options-changes/index.md]: Thanks John, sounds like this is something user can ignore during upgrade (e.g. have a better default in 1.2)

File: content/docs/reference/config/installation-options-changes/index.md [hunk 8]
```diff
@@ -17,443 +17,225 @@ The tables below show changes made to the installation options used to customize
...
+| `global.proxy.resources.limits.memory` | `128Mi` | `1024Mi` |  |  |
+| `global.proxy.dnsRefreshRate` | `5s` | `300s` | `Configure the DNS refresh rate for Envoy cluster of type STRICT_DNS 5 seconds is the default refresh rate used by Envoy`  | `Configure the DNS refresh rate for Envoy cluster of type STRICT_DNS This must be given it terms of seconds. For example, 300s is valid but 5m is invalid.` |
 
-### Modified `security` key/value pairs
+### Modified `mixer` key/value pairs
 
 | Key | Old Default Value | New Default Value | Old Description | New Description |
 | --- | --- | --- | --- | --- |
-| `security.selfSigned` | `true # indicate if self-signed CA is used.` | `true` |  | `indicate if self-signed CA is used.` |
+| `mixer.adapters.useAdapterCRDs` | `true` | `false` | `Setting this to false sets the useAdapterCRDs mixer startup argument to false`  | `Setting this to false sets the useAdapterCRDs mixer startup argument to false` |
```
Comment (linsun) [inline content/docs/reference/config/installation-options-changes/index.md]: @mandarjog @douglas-reid what is the impact of this value during upgrade?  I assume user needs to set this to true if mixer policy is used?

**Comment by pitlv2109** [comment]
@johnma14 There are also a few linter issues that need to be addressed :)

File: content/docs/reference/config/installation-options-changes/index.md [hunk 9]
```diff
@@ -17,443 +17,225 @@ The tables below show changes made to the installation options used to customize
...
 | --- | --- | --- |
-| `global.hyperkube.tag` | `v1.7.6_coreos.0` |  |
-| `global.k8sIngressHttps` | `false` |  |
-| `global.crds` | `true` |  |
-| `global.hyperkube.hub` | `quay.io/coreos` |  |
-| `global.meshExpansion` | `false` |  |
-| `global.k8sIngressSelector` | `ingress` |  |
-| `global.meshExpansionILB` | `false` |  |
+| `global.proxy.tracer` | `"zipkin"` | `Specify which tracer to use. One of: lightstep, zipkin`  |
+| `global.controlPlaneSecurityEnabled` | `false` | `controlPlaneMtls enabled. Will result in delays starting the pods while secrets are propagated, not recommended for tests.`  |
```
Comment (linsun) [inline content/docs/reference/config/installation-options-changes/index.md]: This also still exists... hope it'd be fixed with the tracer.

**Comment by linsun** [comment]
Thank you @johnma14 this look a lot better!

I don't see big issues with install option changes... certainly user need to update to use the new values and work with the deprecated value as needed.

The big thing i think worth call out in relnote is mixer.adapters.useAdapterCRDs, along with new values introduced.

cc @duderino @francoispqt @sdake

File: content/docs/reference/config/installation-options-changes/index.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (johnma14) [reply content/docs/reference/config/installation-options-changes/index.md]: Fixed. This should not show now

File: content/docs/reference/config/installation-options-changes/index.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (johnma14) [reply content/docs/reference/config/installation-options-changes/index.md]: This should be fixed and should not show as a change. Only the description field had changed for this option.

File: content/docs/reference/config/installation-options-changes/index.md (hunk 9 above)
Comment (johnma14) [reply content/docs/reference/config/installation-options-changes/index.md]: Should be fixed now

File: content/docs/reference/config/installation-options-changes/index.md [hunk 10]
```diff
@@ -17,443 +17,225 @@ The tables below show changes made to the installation options used to customize
...
-| `tracing.replicaCount` | `1` |  |
-| `tracing.jaeger.ingress` | `jaeger.local` |  |
-| `tracing.ingress` | `tracing.local` |  |
-| `tracing.jaeger` | `jaeger.local` |  |
-| `tracing` | `jaeger.local tracing.local` |  |
-| `tracing.jaeger.ingress.hosts` | `jaeger.local` |  |
-| `tracing.jaeger.ingress.enabled` | `false` |  |
-| `tracing.ingress.hosts` | `tracing.local` |  |
-| `tracing.jaeger.ui.port` | `16686` |  |
+| `gateways.istio-ingressgateway.resources` | `{}` |  |
```
[the remark this answers is not in the document]
Comment (johnma14) [reply content/docs/reference/config/installation-options-changes/index.md]: Should be fixed now.

File: content/docs/reference/config/installation-options-changes/index.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (johnma14) [reply content/docs/reference/config/installation-options-changes/index.md]: Sounds good.

File: content/docs/reference/config/installation-options-changes/index.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (johnma14) [reply content/docs/reference/config/installation-options-changes/index.md]: So if you look at https://github.com/istio/istio.io/pull/4408/files#diff-d88b59ce29807bd5e0d25dcdcd678f46L49, you will see that in 1.1, it was called `gateways.istio-ingressgateway.resources` and in 1.2 it is called `gateways.istio-ingressgateway.resources.{requests/limits}/{cpu/memory}. Please let me know if my understanding is wrong. Appreciate it @howardjohn

**Comment by johnma14** [comment]
@pitlv2109 the linter failures seems to be on the chinese files. I am not sure how to fix those. Any ideas?

**Comment by johnma14** [comment]
@howardjohn @geeknoid @linsun @sdake - pushed another patchiest out for review. Thanks

File: content/docs/reference/config/installation-options-changes/index.md (hunk 8 above)
Comment (douglas-reid) [reply content/docs/reference/config/installation-options-changes/index.md]: you do not need to set this to `true` if mixer policy is used. this has been an option for at least one release now -- and we have publicly announced the deprecation policy around this shift. If anyone is still using old style Mixer config, they will either need to update this flag at install time or they will need to update their Mixer config to the new style. The old style represents a large burden on the API server which was causing issues in different install environments.

**Comment by sdake** [comment]
@johnma14 - To fix many of the gate failures, I'd highly recommend doing a force rebase.

To do this, the steps I follow are as follows
```
git fetch upstream
git rebase upstream/master
git push -f origin
```

This assumes you have an upstream remote, which in order to push, you probably do.

Cheers
-steve

File: content/docs/reference/config/installation-options-changes/index.md [hunk 11]
```diff
@@ -1,13 +1,13 @@
...
+description: Details the Helm chart installation options differences between release-1.1 and release-1.2.
 weight: 30
 keywords: [kubernetes, helm, install, options]
 ---
 
-The tables below show changes made to the installation options used to customize Istio install using Helm between release 1.0 and release 1.1. The tables are grouped in to three different categories:
+The tables below show changes made to the installation options used to customize Istio install using Helm between release 1.1 and release 1.2. The tables are grouped in to three different categories:
 
-- The installation options already in the previous release but whose values or descriptions have been modified in the new release.
+- The installation options already in the previous release but whose values have been modified in the new release.
```
Comment (rcaballeromx) [inline content/docs/reference/config/installation-options-changes/index.md]: @johnma14 Are you no longer including the options where the descriptions changed? Why?

File: content/docs/reference/config/installation-options-changes/index.md (hunk 8 above)
Comment (rcaballeromx) [reply content/docs/reference/config/installation-options-changes/index.md]: @johnma14 @linsun The description does not state what the key/value pair does but rather the effect it has on a different key/value pair. Could we include both pieces of information?

File: content/docs/reference/config/installation-options-changes/index.md [hunk 12]
```diff
@@ -17,443 +17,217 @@ The tables below show changes made to the installation options used to customize
...
-| --- | --- | --- |
-| `istio_cni.enabled` | `false` |  |
-
-### New `servicegraph` key/value pairs
+### New `tracing` key/value pairs
 
 | Key | Default Value | Description |
 | --- | --- | --- |
-| `servicegraph.nodeSelector` | `{}` |  |
+| `tracing.podAntiAffinityLabelSelector` | `[]` |  |
```
Comment (rcaballeromx) [inline content/docs/reference/config/installation-options-changes/index.md]: @johnma14 @linsun Please include descriptions.

File: content/docs/reference/config/installation-options-changes/index.md (hunk 11 above)
Comment (johnma14) [reply content/docs/reference/config/installation-options-changes/index.md]: @rcaballeromx  If values remain the same and only the descriptions change, we decided not to track that as a change here(You can see discussion around this in: https://github.com/istio/istio.io/pull/4402. This was done to put focus on the changes to the values and avoid the noise around documenting these options). This file will be used only to track changes in the values of configuration options, new options added and old ones that are removed. Hope this helps.

File: content/docs/reference/config/installation-options-changes/index.md (hunk 2 above)
Comment (johnma14) [reply content/docs/reference/config/installation-options-changes/index.md]: @rcaballeromx we would have to add a description to the values.yaml file of the corresponding options. Not sure if we have enough time to get these added in for 1.2. If we want to do this for this release, please let me know.

File: content/docs/reference/config/installation-options-changes/index.md (hunk 8 above)
Comment (johnma14) [reply content/docs/reference/config/installation-options-changes/index.md]: @rcaballeromx we would have to change this description in the values.yaml file for mixer. Not sure if we want to do this for 1.2. If we want, please let me know and I will make the changes.

File: content/docs/reference/config/installation-options-changes/index.md [hunk 13]
```diff
@@ -17,443 +17,217 @@ The tables below show changes made to the installation options used to customize
 
 ## Modified configuration options
 
-### Modified `servicegraph` key/value pairs
+### Modified `kiali` key/value pairs
 
 | Key | Old Default Value | New Default Value | Old Description | New Description |
 | --- | --- | --- | --- | --- |
-| `servicegraph.ingress.hosts` | `servicegraph.local` | `servicegraph.local` |  | `Used to create an Ingress record.` |
+| `kiali.hub` | `docker.io/kiali` | `quay.io/kiali` |  |  |
```
[the remark this answers is not in the document]
Comment (johnma14) [reply content/docs/reference/config/installation-options-changes/index.md]: @rcaballeromx We parse the values.yaml file to get the description to populate the Description field. If there are no descriptions, we leave this field empty. There are also some values.yaml file where some of the configuration options are commented out. Since we do not want those to be included in the descriptions of the following configuration option, we have filters in place to see if a comment field has a format: <something>: <something>, then we ignore those descriptions (to be careful that we do not include comments that might contain commented out options.) So these values.yaml files need some cleanup but we refrained from doing that since @ostromart is working on developing a cleaner approach to get the schema of these options which also includes providing descriptions for these options.

File: content/docs/reference/config/installation-options-changes/index.md (hunk 12 above)
Comment (sdake) [reply content/docs/reference/config/installation-options-changes/index.md]: @rcaballeromx it will take weeks to add these description fields.  These files are autogenerated from this files:

https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/values.yaml

as well as all of the `values.yaml` in the subcharts.

This work would take weeks, and i don't think we want to hold the release for the lack of description fields, as we would have to spin another RC - and potentially delay the release.

**Comment by johnma14** [comment]
@linsun @howardjohn @duderino @sdake @rcaballeromx This PR is ready for review. The following changes should be reflected in the latest version:
1)  Ran this against 1.2.0-rc.3 -
In the installation-option-changes file, 2 new option changes got reflected under the Newly added options: mixer.template.useTemplateCRD and security.citadel.HealthCheck
2) Another new option was added: global.imagePullSecrets - 
This is related to this PR: https://github.com/istio/istio.io/pull/4446

Thanks.

File: content/docs/reference/config/installation-options-changes/index.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (rcaballeromx) [reply content/docs/reference/config/installation-options-changes/index.md]: @johnma14 could you file an issue referencing this PR for us to follow up after release?
You can base it on this example #4478

**Comment by rcaballeromx** [comment]
@johnma14 Please rebase your changes. That should help you solve the tests errors. Please make sure that your changes are rendered correctly.

**Comment by johnma14** [comment]
@rcaballeromx I did a rebase but circleci is still failing. I verified the docs are rendering correct.Thanks.

File: content/docs/reference/config/installation-options-changes/index.md (hunk 2 above)
Comment (linsun) [reply content/docs/reference/config/installation-options-changes/index.md]: I've called this out in relnote if you want a reference.

File: content/docs/reference/config/installation-options-changes/index.md (hunk 8 above)
Comment (linsun) [reply content/docs/reference/config/installation-options-changes/index.md]: The details are highlighted in upgrade notice by @sdake

File: content/docs/reference/config/installation-options-changes/index.md (hunk 12 above)
Comment (linsun) [reply content/docs/reference/config/installation-options-changes/index.md]: I think this is okay - we did add a link to k8s in release note for this.  This is well known for folks know k8s well.

**Comment by linsun** [comment]
Oh rebase is well needed, it will help you @johnma14 resolve these lint errors, which did for me.

**Comment by geeknoid** [comment]
Based on Lin's recommendation, merging this.
